### PR TITLE
Enumerable Methods

### PR DESCRIPTION
**Pull request by FrancoRosa**
In this project:
I built the following methods:
Inside the Enumerable Module
> my_each
> my_each_with_index
> my_select
> my_all?
>  my_any?
> my_none?
> my_count
> my_map
> my_inject

Finally, I made a simple function using _my_inject_

> multiply_els

### CODE REVIEW

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @FrancoRosa       
#### Great Job so far, but there are a few more things required. 
### General
- [x] Lets fix stickler errors
- [x] Pull Request should have a meaningful title

- [x] `my_count`
-  returns the number of items in enum through enumeration
`p [8,87,6,5].my_count` should return `4`
- counts the number of items in enum that are equal to item if an argument is given:
`p  [1, 2, 3].my_count(2)` should return `1`
-  counts the number of elements yielding a true value if a block is given
`block = proc { |num| num < (2 + 24) / 2 }`
`p [23,13,12,43].my_count(&block)` should return 1

- [x] `my_inject`:
- Combines all elements of enum by applying a binary operation, specified by a block:
`search = proc { |memo, word| memo.length > word.length ? memo : word } `
`p ['hello','strong','am'].inject(&search)` should return `strong`
- [x] `my_none?`:
- When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello'].my_none?(5)` - should return `true`
- [x] `my_any?`:
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
`p ['cat',2,32,].my_any?('cat')` should return `true`
- [x] `my_all`  
- when no block or argument is given returns true when none of the collection members are false or nil:
 `p [1, true, 'hi', []].my_all?` should return `true`.
- when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`p [3,3,3,3].my_all?(3)` should return `true`



### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @FrancoRosa     
#### Great Job so far, you are almost there .
### General

- [x] `my_all`  
- when no block or argument is given returns true when none of the collection members are false or nil:
`p [1, true, 'hi',false].my_all?` should return `false`
- [x] `my_inject`:
- When a symbol is specified combines each element of the collection by applying the symbol as a named method
`p (5..10).my_inject(:*)` should return `151200`

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by pelzolga123** [comment]
## Hi Franco,
### Good job so far :+1:  just a few changes left: 

* [ ] `[1, true, 'hi', []].my_all?` should return the same result as original ruby method: `[1, true, 'hi', []].all?` 
* [x] Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+AllCops:
+  Exclude:
+    - "README.md"
+    - "Guardfile"
+    - "Rakefile"
+    - "tests.rb"
```
Comment (pelzolga123) [inline .rubocop.yml]: Do not make any changes in config files.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  DisplayCopNames: true
+
+Layout/LineLength:
+  Max: 120
+Metrics/MethodLength:
+  Max: 30
```
Comment (pelzolga123) [inline .rubocop.yml]: Do not make any changes in config files

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Metrics/AbcSize:
+  Max: 50
+Metrics/ClassLength:
+  Max: 180
```
Comment (pelzolga123) [inline .rubocop.yml]: Do not make any changes in config files

File: scripts.rb [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    mem = arr.shift
+    if args[0].is_a?(Numeric)
+      arr.unshift(args[0])
+      if !args[1]
+        arr.my_each { |k| mem = yield(mem, k) }
+      else
+        arr.my_each { |k| mem = mem.send(args[1], k) }
+      end
+    elsif args[0].is_a?(Symbol)
+      arr.my_each { |k| mem = mem.send(args[0], k) }
```
Comment (majovanilla) [inline scripts.rb:127]: This piece of code is not working. When I run `[1,2,3,4,5].my_inject(:+)` on your project, I get the wrong sum!

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey @FrancoRosa , 👋

## You did an *amazing*🌟 ✨ job on this project! 👊🏻  💯

## You just have ONE thing to change and submit your request again! 😀

Please check the comment below on you code.

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @FrancoRosa      
#### Great Job so far, everything works well but i cant approve this project until everything is fine with the stickler  
### General
- [ ] Please fix the stickler 
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by shshamim63** [comment]
## Status :100:  Approved :100: 
___
Hi @FrancoRosa :wave: ,
Awesome job :clap:, You have covered all the changes asked by the previous reviewer. And you have changed the rubocop.yml file cause of the new release, I understand. But it is always better to add comments if you need to update the crucial files like stickler and robocop.
____
##### This :+1:  is from me. You can go ahead and merge it now.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Twitter](https://twitter.com/Shshamim090)

**Pull request merged**